DOC: Proofread tunnel.rst (#1753)
vyos/vyos-documentation: docs/configuration/interfaces/tunnel.rst

:lastproofread: 2026-01-23

.. _tunnel-interface:

Tunnel
======

Tunnel interfaces are virtual interfaces that transmit encapsulated traffic 
between private networks or hosts across public infrastructure, such as the 
Internet. They operate using encapsulation protocols, such as :abbr:`GRE 
(Generic Routing Encapsulation)` or IPIP, which define how the original traffic 
is encapsulated for transport.

While :abbr:`GRE (Generic Routing Encapsulation)` is often the preferred 
one-size-fits-all solution due to its versatility, VyOS also supports other 
tunnel interface types that may be better suited for specific use cases. 

> The supported tunnel interfaces and their configuration options are described 

Reviewer: This wording obscures that fact that there are no separate interface types for GRE, IPIP, IP6IP6, etc. tunnels — all those protocols are configured using `set interfaces tunnel` commands at the moment. The message here is "there's no separate GRE interface type, it's all under tunnel".

Author: Corrected.

IPIP
----

IPIP is a straightforward encapsulation protocol defined in RFC 2003. It 
encapsulates one IPv4 packet inside another IPv4 packet. 

IPIP tunnel interfaces have very few configuration options and require minimal 

> CPU and memory to operate. They are typically used for simple point-to-point 

Reviewer: I don't see a reason to mention CPU and memory here. None of those protocols are especially CPU- or memory-intensive to process.

Author: Corrected.

IP6IP6
------

IP6IP6 is the IPv6 counterpart to IPIP. It encapsulates one IPv6 packet inside 
another IPv6 packet.

Similar to their IPIP counterparts, IP6IP6 tunnel interfaces have very few 

> configuration options and require minimal CPU and memory to operate. 

Reviewer: Did the CPU and memory bit come from an LLM, I wonder?

Author: Removed.

IPIP6
-----

IPIP6 is an encapsulation protocol that wraps IPv4 packets inside IPv6 packets.

Similar to those of IPIP and IP6IP6, IPIP6 tunnel interfaces are easy to 

> configure and require minimal CPU and memory. They are ideal for unencrypted 

Reviewer: Another point here is that those tunnels are often used in conjunction with IPSec — the tunnel provides encapsulation and the fact that it's a separate virtual interface makes it easy to manage firewall rules and similar, and IPsec provides security. We may want to mention that none of these protocols are encrypted somewhere in the page but I don't see a reason to mention that for individual protocols.

Author: Removed.

6in4 (SIT)
----------

6in4, also known as :abbr:`SIT (Simple Internet Transition)`, is an 
encapsulation protocol defined in :rfc:`4213` that wraps IPv6 packets 
inside IPv4 packets. The encapsulating IPv4 headers use IP protocol number 41, 
which is reserved exclusively for IPv6 encapsulation.

The encapsulation process adds a 20-byte IPv4 header to each IPv6 packet. 
Consequently, 6in4 tunnel interfaces can transmit IPv6 packets up to 1480 bytes 
over an underlying network with a standard MTU of 1500 bytes without 
fragmentation. 

6in4 tunnel interfaces are frequently used by IPv6 tunnel brokers (such as 

> `Hurricane Electric`_) to connect isolated IPv6 networks or individual hosts to 

Reviewer: Side note: https://tunnelbroker.net/ is still operational! I was wondering if this bit had become outdated, it has not.

Author: The reference is preserved.

  set interfaces tunnel tun0 remote 192.0.2.20
  set interfaces tunnel tun0 address 2001:db8:bb::1/64

.. seealso:: For a practical configuration example, see the 
   :ref:`Tunnelbroker.net (IPv6) <examples-tunnelbroker-ipv6>` section.

Generic Routing Encapsulation (GRE)
-----------------------------------

:abbr:`GRE (Generic Routing Encapsulation)` is a versatile encapsulation 
protocol defined in RFC 2784. Unlike simpler protocols such as IPIP, it allows 

> IPv4, IPv6, multicast, and other traffic types to be transported through the 

Reviewer: IPIP doesn't break multicast traffic in any way, nor does GRE offer anything special for multicast. There's nothing special about multicast at the tunneling protocol level — a multicast packet is just a packet with a destination address from a subnet designated for multicast.

```suggestion
both IPv4 and IPv6 to be transported through the 
```

same tunnel.

:abbr:`GRE (Generic Routing Encapsulation)` encapsulates original data packets 
by adding a :abbr:`GRE (Generic Routing Encapsulation)` header, followed by an 
IP header (the delivery header). The delivery header uses IP protocol number 47 
to identify :abbr:`GRE (Generic Routing Encapsulation)`-encapsulated traffic.

In VyOS, :abbr:`GRE (Generic Routing Encapsulation)` tunnels can be established 
over both IPv4 (encapsulation ``gre``) and IPv6 (encapsulation ``ip6gre``) 
transport networks.

Configuration
^^^^^^^^^^^^^

To configure a :abbr:`GRE (Generic Routing Encapsulation)` tunnel, you need to 
define a tunnel source IP address, a tunnel destination IP address, an 
encapsulation type (:abbr:`GRE (Generic Routing Encapsulation)`), and a tunnel 
interface IP address.

Example:

The following example shows how to configure an IPv4-over-IPv4 :abbr:`GRE 
(Generic Routing Encapsulation)` tunnel between a VyOS router and a Cisco IOS 
router. The primary difference between the two platforms is that VyOS requires 
the encapsulation type to be explicitly defined, while Cisco IOS uses 

> :abbr:`GRE (Generic Routing Encapsulation)` over IP by default.

Reviewer: I wonder if that's still true in recent IOS-XE versions.

Author: This example has been removed from the user guide and will be migrated to the KB.

  set interfaces tunnel tun100 source-address '198.51.100.2'
  set interfaces tunnel tun100 remote '203.0.113.10'

> **Cisco IOS router:**

Reviewer: I wonder if these configuration examples should be moved to the knowledge base right away, since they are beyond the scope of the VyOS reference manual.

Author: This example has been removed from the user guide and will be migrated to the KB.

.. code-block:: none

   set interfaces tunnel tun1 source-address 192.0.2.10
   set interfaces tunnel tun1 remote 192.0.2.20
   set interfaces tunnel tun1 address 172.16.17.18/24
   set interfaces tunnel tun1 parameters ip key 20

GRETAP
^^^^^^^

Unlike GRE, which encapsulates only Layer 3 (IP) traffic, GRETAP encapsulates 
Layer 2 (Ethernet) frames.

Because it preserves MAC addresses, GRETAP tunnel interfaces are often added to 
a network bridge. This allows two geographically distant sites to connect as if 

> they were on the same LAN.

Reviewer: ```suggestion
That means that GRETAP tunnel interfaces can be members of a bridge interface. This allows two geographically distant sites to connect as if 
they were on the same LAN.
```

It can be added to a bridge because it _encapsulates Ethernet frames_. That's the necessary and sufficient conditions, everything else is simply irrelevant.

Also, "often" is subjective. The key information is that it's possible, and that's what the docs should say.

Troubleshooting
^^^^^^^^^^^^^^^

GRE is a standardized tunneling protocol used in many network environments.

Although the GRE tunnel setup is straightforward, its successful deployment 

> requires proper alignment with the underlying network security policies.

Reviewer: Please do not remove useful information. "Check that protocol 47" is allowed is useful and actionable. "Proper alignment with the underlying security policies" is just a bunch of empty words.

Author: Corrected.
